tools/importer-rest-api-specs/components/dataapigeneratorjson/templater_terraform_resource_mappings.go

```diff
@@ -0,0 +1,94 @@
+package dataapigeneratorjson
+
+import (
+	"encoding/json"
+	"github.com/hashicorp/pandora/tools/sdk/resourcemanager"
+)
+
+type ResourceMapping struct {
+	ResourceIdMapping             *[]ResourceIdMapping            `json:"ResourceIdMapping"`
+	DirectAssignmentFieldMappings *[]DirectAssignmentFieldMapping `json:"DirectAssignmentFieldMappings,omitempty"`
+	ModelToModelFieldMappings     *[]ModelToModelFieldMapping     `json:"ModelToModelFieldMappings,omitempty"`
+}
+
+type ResourceIdMapping struct {
+	MappingFunction string
+	SegmentName     string
+}
+
+type DirectAssignmentFieldMapping struct {
+	SchemaModelName string `json:"SchemaModelName"`
+	SchemaFieldPath string `json:"SchemaFieldPath"`
+	SdkModelName    string `json:"SdkModelName"`
+	SdkFieldPath    string `json:"SdkFieldPath"`
+}
+
+type ModelToModelFieldMapping struct {
+	SchemaModelName string `json:"SchemaModelName"`
+	SdkModelName    string `json:"SdkModelName"`
+	SdkFieldName    string `json:"SdkFieldName"`
+}
+
```

Comment:
I think there might be one more struct missing for `ModelToModelMapping`

```suggestion
type ModelToModelMapping struct {
	SchemaModelName string `json:"SchemaModelName"`
	SdkModelName string `json:"SdkModelName"`
}
```

Reply:
Fixed and added DirectAssignmentFieldMapping/ModelToModelFieldMapping as they are slightly different and it's probably worth adding that distinction

Reply:
I'm not sure if there are some commits missing, but I don't see them

Reply:
Not sure what happened here, but these are above now.